.github/action_scripts/send_transactions/allow-spends-and-spend-transactions.js

```diff
@@ -0,0 +1,422 @@
+        const activeAllowSpendsForAddress = isCurrency
+            ? snapshot[1]?.activeAllowSpends?.[address]
+            : snapshot[1]?.activeAllowSpends?.[tokenId]?.[address];
+
+        if (!activeAllowSpendsForAddress) {
+            throw new Error(`No active allow spends found for address ${address} in ${layerName}`);
+        }
+
+        const hasMatchingHash = await findMatchingHash(activeAllowSpendsForAddress, hash);
+        if (!hasMatchingHash) {
+            throw new Error(`Allow spend with hash ${hash} not found in ${layerName}`);
+        }
+
+        console.log(`AllowSpend transaction found in ${layerName}`);
+    };
+
+    const verifyInL0 = async (address, hash, l0Url, tokenId, layerName, isCurrency = false) => {
+        await withRetry(
+            async () => {
+                const { data: snapshot } = await axios.get(
+                    `${l0Url}/global-snapshots/latest/combined`
+                );
+                await verifyAllowSpendInSnapshot(address, hash, snapshot, tokenId, `${layerName} L0`, isCurrency);
+            },
+            { name: `${layerName} L0 verification` }
+        );
+    };
+
+    const verifyInCurrencyL0 = async (address, hash) => {
+        await withRetry(
+            async () => {
+                const { data: snapshot } = await axios.get(
+                    `${urls.currencyL0Url}/snapshots/latest/combined`
+                );
+                await verifyAllowSpendInSnapshot(
+                    address, 
+                    hash, 
+                    snapshot,
+                    CONSTANTS.CURRENCY_TOKEN_ID,
+                    'Currency L0',
+                    true
+                );
+            },
+            { name: 'Currency L0 verification' }
+        );
+    };
+
+    return {
+        verifyDagL1: (hash) => verifyInL1(hash, urls.dagL1Url, 'DAG'),
+        verifyDagL0: (address, hash) => verifyInL0(address, hash, urls.globalL0Url, '', 'DAG', false),
+        verifyCurrencyL1: (hash) => verifyInL1(hash, urls.currencyL1Url, 'Currency'),
+        verifyCurrencyL0: (address, hash) => verifyInCurrencyL0(address, hash),
+        verifyGlobalL0ForCurrency: (address, hash) => 
+            verifyInL0(address, hash, urls.globalL0Url, CONSTANTS.CURRENCY_TOKEN_ID, 'Global', false)
+    };
+};
+
+const createBalanceManager = (urls) => {
+    const getBalance = async (privateKey, l0Url, l1Url, isCurrency = false) => {
+        const account = dag4.createAccount(privateKey);
+        const address = account.address;
+        
+        try {
+            const snapshotUrl = isCurrency 
+                ? `${l0Url}/snapshots/latest/combined`
+                : `${l0Url}/global-snapshots/latest/combined`;
+            
+            const { data: snapshot } = await axios.get(snapshotUrl);
+            
+            const balance = isCurrency
+                ? snapshot[1]?.balances?.[address] || 0
+                : snapshot[1]?.balances?.[address] || 0;
```

Comment:
Both cases look the same. Also is the fallback to `0` correct? I mean can we have a situation that request passes but we will get `null` there?

Reply:
Ah, you're right. I changed one place and missed the other